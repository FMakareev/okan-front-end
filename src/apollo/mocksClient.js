import faker from 'faker';
import {GraphQLError} from 'graphql';

import setupClient from './helpers/apolloClientMock';

import schema from './schema.graphqls';

import {userlist} from './graphql/query/userlist';
import {useritem} from './graphql/query/useritem';

import {ROLE_ADMIN, ROLE_USER} from '../shared/roles';
import {celItem} from "./graphql/query/celItem";
import cellTree from './graphql/query/cellTree';
import {documentitem} from "./graphql/query/documentItem";
import {documentlist} from "./graphql/query/documentlist";
import {projectitem} from "./graphql/query/projectitem";

const defaultMocks = {
  Query: () => ({
    userlist: () => userlist(faker.random.number(5)),
    useritem,
    documentitem:()=>documentitem(),
    projectitem:(query, {id})=>{
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve(projectitem({id}));
        }, faker.random.number(0))
      })
    },
    documentlist:()=>{
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve(documentlist(1));
        }, faker.random.number(0))
      })
    },
    cellitem: (query, props) => {

      console.log('cellitem: ', props);
      const {id, prevcell, nextcell, parent} = props;
      return new Promise((resolve, reject) => {

        setTimeout(() => {
          const result = cellTree.find(item => item.id === id);

          console.log('cellitem result: ', result);
          resolve(result);
        }, faker.random.number(0))
      })
    },
    useremailitem: (query, {email}) => {
      switch (email) {
        case 'client@okan.su': {
          return {...useritem(), email: 'client@okan.su', role: ROLE_USER};
        }
        case 'admin@okan.su': {
          return {...useritem(), email: 'admin@okan.su', role: ROLE_ADMIN};
        }

        default: {
          // throw new GraphQLError('user not found');
          throw Error(
            JSON.stringify({
              useremailitem: null,
              errors: [
                {
                  message: 'GraphQL error: user not found',
                },
              ],
            }),
          );
        }
      }
    },
  }),

  Mutation: () => ({
    /**
     * @params {func} mutation - этот же запрос
     * @params {object} props - аргументы которые были переданы
     * */
    createuser: (mutation, props) => {
      // TODO review: добавить рандомную задержку от 500 до 3000
      return setTimeout(() => {
        if (props.email === 'error@okan.su') {
          throw new GraphQLError('already registered');
        } else {
          return props;
        }
      }, 10000);
    },

    activateuser: (mutation, props) => {
      // TODO review: добавить рандомную задержку от 500 до 3000
      return setTimeout(() => {
        if (props.email === 'error@okan.su') {
          throw new GraphQLError('already registered');
        } else {
          return props;
        }
      }, 5000);
    },

    updateuser: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve({...useritem()})
            : reject(
            JSON.stringify({
              errors: [
                {
                  message: 'error!',
                },
              ],
            }),
            );
        }, faker.random.number(2000));
      }),

    changepassword: (mutation, props) => props,

    createcell: (mutation, props) => {

      console.log('cellitem: ', props);
      const {prevcell, parent} = props;
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve({
            ...celItem({
              prevcell,
              parent
            }),
          });
        }, faker.random.number(2000))
      })
    },
  }),
};

export const mocksClient = setupClient(defaultMocks, schema)();

export default mocksClient;
