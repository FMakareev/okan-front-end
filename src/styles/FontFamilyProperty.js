/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
