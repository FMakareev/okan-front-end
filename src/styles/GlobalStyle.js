import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
