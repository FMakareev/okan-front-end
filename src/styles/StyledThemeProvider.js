import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
