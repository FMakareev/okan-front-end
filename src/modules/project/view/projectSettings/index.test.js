import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider as ProviderRedux } from 'react-redux';
import { ApolloProvider } from 'react-apollo';
import { createStore } from 'redux';

import { ProjectSettingsPage } from './index';
import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';
import mocksClient from '../../../../apollo/mocksClient';

const store = createStore(() => {});

test('ProjectSettingsPage: рендер без ошибок', () => {
  const renderWithProps = props => {
    const defaultProps = {
      match: {
        params: { key: '3c64e11c68c5159fb2a87b89ed719ae900e90ff77e2e0b19426b950b4fe8ff33' },
      },
    };
    return renderer.create(
      <StyledThemeProvider>
        <BrowserRouter>
          <ProviderRedux store={store}>
            <ApolloProvider client={mocksClient}>
              <ProjectSettingsPage {...defaultProps} />
            </ApolloProvider>
          </ProviderRedux>
        </BrowserRouter>
      </StyledThemeProvider>,
    );
  };
  expect(renderWithProps({})).toMatchSnapshot();
});
