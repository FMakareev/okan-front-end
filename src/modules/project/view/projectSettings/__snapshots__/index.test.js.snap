// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSettingsPage: рендер без ошибок 1`] = `
<div
  className="sc-bxivhb gPUjkv"
>
  <div
    className="sc-1jxwge6-0 wterhv-0 bLkhJm"
  >
    <div
      className="sc-1jxwge6-0 sc-14k9gvj-0 jTXbMH"
      width="100%"
    >
      Загрузка ...
    </div>
  </div>
</div>
`;
